### PR TITLE
[Swork] Reconstruct swork services

### PR DESCRIPTION
**Pull request by yuhui1208**

### CODE REVIEW

File: src/services/util.ts [hunk 1]
```diff
@@ -14,6 +20,64 @@ export function loadKeyringPair(api: ApiPromise, req: Request): KeyringPair {
...
+
+              console.log(`✅ Send transaction(${tx.type}) success.`);
+              resolve(result);
+            }
+          });
+        }
+      });
+    } catch (error) {
+      reject(error);
+      throw error;
```
Comment (pangwa) [inline src/services/util.ts:69]: throw is not necessary.

File: src/services/util.ts [hunk 2]
```diff
@@ -14,6 +20,64 @@ export function loadKeyringPair(api: ApiPromise, req: Request): KeyringPair {
...
+export async function sendTx(tx: SubmittableExtrinsic, krp: KeyringPair) {
+  return new Promise((resolve, reject) => {
+    try {
+      tx.signAndSend(krp, ({events = [], status}) => {
+        console.log('💸 Transaction status:', status.type);
+
+        if ('Invalid' === status.type) {
+          reject(new Error('Invalid transaction.'));
+        }
+        if (status.isInBlock) {
```
Comment (pangwa) [inline src/services/util.ts:32]: should handle "else"

File: src/services/index.ts [hunk 3]
```diff
@@ -34,13 +102,50 @@ export const block = {
 };
 
 export const swork = {
-  register: (req: Request, res: Response, next: any) => {
+  register: (req: Request, res: Response, next: NextFunction) => {
+    api.isReady.then(async api => {
```
Comment (pangwa) [inline src/services/index.ts:106]: looks like the api.isReady.then... is a common block. A function could be extracted like:
```javascript
async function withApi(fn, next) {
 const api2 = await api.isReady;
  try {
    await fn(api2);
  } catch(ex) {
  next(ex);
  }
}

// usage:
withApi(async (api) => {
  const krp = loadKeyringPair(req);
  res.json(await register(api, krp, req); // the res.json part could be placed in the withApi function if necessary.
});
```

File: src/services/swork.ts [hunk 4]
```diff
@@ -1,35 +1,82 @@
...
+  return await sendTx(tx, krp);
+}
+
+/**
+ * Queries
+ */
+export async function identity(api: ApiPromise, req: Request) {
+  const address = req.query['address'];
+  const pks: string[] = queryToObj(await api.query.swork.idBonds(address));
+  return await Promise.all(
```
Comment (pangwa) [inline src/services/swork.ts:59]: consider adding limiting max number of parallels calls here?

**Review by pangwa — APPROVED** [review]

**Review by pangwa — APPROVED** [review]

File: src/services/swork.ts (hunk 4 above)
Comment (zikunfan) [reply src/services/swork.ts:61]: Add TODO cause currently we only has 10 wr

**Pull request merged**